File: src/test/fixtures/typescript/logicalControlFlow.ts
// clean — not flagged by logical control flow
function cleanExplicitIf(isLoggedIn: boolean) {
    if (isLoggedIn) {
        navToDashboard();
    }
}

// flagged — logical control flow (low)
function flaggedAndAsIf(isLoggedIn: boolean) {
    isLoggedIn && navToDashboard();
}

// flagged — logical control flow (low)
function flaggedOrAsUnless(isLoggedIn: boolean) {
    isLoggedIn || navToDashboard();
}

File: src/test/fixtures/cpp/logical_control_flow.cpp
// clean — not flagged by logical control flow
void cleanExplicitIf(bool isLoggedIn) {
    if (isLoggedIn) {
        navToDashboard();
    }
}

// flagged — logical control flow (low)
void flaggedAndAsIf(bool isLoggedIn) {
    isLoggedIn and navToDashboard();
}

// flagged — logical control flow (low)
void flaggedOrAsUnless(bool isLoggedIn) {
    isLoggedIn or navToDashboard();
}
